### PR TITLE
feat/user-login-aabha

### PR DESCRIPTION
**Pull request by xentro-sam**

### CODE REVIEW

**Review by Saatwik-S — APPROVED** [review]

File: src/util/abdm/userOnboardingApi.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    ...userData,
+    ...userDetails,
+    txnId
+  }, {
+    headers: {
+      Authorization: `Bearer ${token}`
+    }
+  })
+  return response
+}
```
Comment (rkdixit96) [inline src/util/abdm/userOnboardingApi.js:64]: Move urls to config file

File: src/util/abdm/publicKeyFetcher.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const axios = require('axios')
+
+const getPublicKey = async (version = 'v1') => {
+  const CERT_URL = `https://healthidsbx.abdm.gov.in/api/${version}/auth/cert`
+  const publicKey = await axios.get(CERT_URL)
+  return publicKey.data
+}
```
Comment (rkdixit96) [inline src/util/abdm/publicKeyFetcher.js:7]: This is a service

File: src/util/abdm/authToken.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    return new Promise((resolve, reject) => {
+      fs.readFile(FILE_NAME, 'utf-8', (error, data) => {
+        error ? reject(error) : resolve(data)
+      })
+    })
+  }
+  return '' // File do not exist
+}
+
+const getJWTToken = async () => {
```
Comment (rkdixit96) [inline src/util/abdm/authToken.js:38]: This is a service

File: src/util/abdm/authToken.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+const jose = require('jose')
+const fs = require('fs')
+const FILE_NAME = 'fetchedToken.txt'
+const tokenFileExists = () =>
+  fs.existsSync(FILE_NAME)
+
+const parseJWT = (jwt) => jose.decodeJwt(jwt)
+
+const fetchJWTFromServer = async () => {
```
Comment (rkdixit96) [inline src/util/abdm/authToken.js:13]: This is a service

**Review by rkdixit96 — CHANGES_REQUESTED** [review]

**Comment by rkdixit96** [comment]
Lets move to use axios redis cache adapter for storing tokens - https://www.npmjs.com/package/axios-cache-adapter.
We'll do this in a separate PR

**Comment by xentro-sam** [comment]
Requested changes made. Kindly approve PR.

**Review by rkdixit96 — APPROVED** [review]

**Pull request merged**